### ISSUE TITLE
Finding the path of the installed hooks

### ISSUE THREAD

**Issue by paw-lu** [pre-commit/pre-commit#1468]
I love the library. Thanks for all of the great work here!

Is there a convenient way to find where the command is located for installed hooks? For some background I'm looking for ways to integrate pre-commit linters and formatters (flake8 and black) into VS Code by [specifying the full path of the linter executable](https://github.com/Microsoft/vscode-python/issues/1707#issuecomment-391153765).

For example, if I use the flake8 hook

```yaml
  - repo: https://gitlab.com/pycqa/flake8
    rev: 3.8.1
    hooks:
      - id: flake8
```

and run `pre-commit install`, once I attempt to commit a file pre-commit will create a virtual environment, install flake8, and run it against the staged file.

Is there a way to find where exactly this flake8 is located? I've found `~/.cache/pre-commit`, but it's not obvious to me which repo and bin correspond to which hook.

Thanks!

**Comment by asottile** [comment]
this implementation is intentionally not exposed -- can you use `pre-commit run flake8 --files` instead?

there's also another vs code plugin by @magicmark iirc

**Comment by paw-lu** [comment]
Thank you for getting back to me @asottile !

Unfortunately, routing thought pre-commit and using `--file` isn't a viable option because of the issue listed in #1233—VS Code will automatically pass a `--format` option to whatever the user specifies as the flake8 command. pre-commit obviously won't recognize that option, so it fails.

Didn't know about the extension thought, thanks for pointing that out! I don't think it will work in my specific case since I'm after the visual indicators in the editor when the code breaks a linting rule.

I'm going to close this, as you essentially answered my question. Again, thanks!

**Comment by playpauseandstop** [comment]
Hi @paw-lu 

I've ran into same issue as you did. 

I would like to run `flake8` (`black`, `mypy`, etc) checks in VS Code (or Sublime Text 3) with `pre-commit`. As `pre-commit` is not designed to do it, I've decided to write my own small library: [pre-commit-run-hook-entry](https://github.com/playpauseandstop/pre-commit-run-hook-entry).

It currently support running black, flake8 & mypy checks via pre-commit hooks for VS Code & Sublime Text 3 editors. Under the hood it done everything, that `pre-commit run ...` does (checking pre-commit config, marking config as used, installing hook env, etc).

To solve your original question next config should be added to your `.vscode/settings.json`,

```json
{
    "python.linting.flake8Enabled": true,
    "python.linting.flake8Path": "pre-commit-run-hook-entry",
    "python.linting.flake8Args": ["flake8"]
}
```
